src/components/collapsible-panel.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+import { caretDownIcon, caretRightIcon } from '@jupyterlab/ui-components';
+
+export interface ICollapsiblePanelProps {
+  title: JSX.Element | string;
+  content: JSX.Element | string;
+  expanded?: boolean;
+}
+
+export function CollapsiblePanel(props: ICollapsiblePanelProps): JSX.Element {
+  const [expanded, setExpanded] = useState(props.expanded ?? false);
+
+  const handleChange = () => {
+    setExpanded(!expanded);
+  };
+
+  return (
+    <div className={'jp-collapsible-panel' + (expanded ? ' expanded' : '')}>
+      <div
```

Comment:
What is the purpose of having a summary container? The way I see this component is just 2 sections, 1) header and 2) body. Header should include the icon and the title, since clicking on either should expand/close. Here is what I propose to simplify this component.

``` html
<div class='jp-collapsible-panel'>
    <div class='jp-collapsible-panel-header' onClick={handleChange}>
        <icon></icon>
        <span class='jp-collapsible-panel-title'>{title}</span>
    </div>
    <div class='jp-collapsible-panel-body'>
        {content}
    </div>
</div>
```

Reply:
Good point, simplified as outlined above and also by using <h2> instead of <div>